### DISCUSSION TITLE
Publishing from Lens Studio to Snap Camera Server

### DISCUSSION THREAD

**Discussion by Hypnotistchris** [ptrumpis/snap-camera-server#13] [Ideas]
Thank you so much for all the work you have done so far.
It currently seems like any Lens created above 4.36.1 will not work on the last version of snap camera. I expect Snapchat will eventually stop new lenses created below version 5 being able to be published.

With this in mind would it be possible to come up with a solution to enable publishing from Lens Studio straight into your Snap Camera Server? It would then completely remove any dependency on snapchat servers.

**Comment by ptrumpis** [comment]
Importing lenses will probably be easier next week as I will update [Snap Lens Cache Import](https://ptrumpis.github.io/snap-lens-cache-import/) so that .zip / .lns files can be imported and not just the cache.

For all new lenses that were created after the shutdown, I have already found a solution that will probably be integrated into the next server version.

You can read about it in the Wiki:
- https://github.com/ptrumpis/snap-camera-server/wiki/Updating-Snap-Camera-(Unofficial)
- https://github.com/ptrumpis/snap-camera-server/wiki/Snap-Camera-Assets

**Reply by Hypnotistchris** [reply]
This is great but being able to publish straight to the server from lens studio instead of publishing to snap chat then downloading it would remove the cloud part completly

**Reply by DeJMan** [reply]
@Hypnotistchris Did you ever find a way to do this?

**Reply by BryanHaley** [reply]
This is likely not easily possible as I don't see any "export" functionality in Lens Studio. Maybe it's possible to intercept the "preview" version being sent to snapchat servers and then your phone for testing purposes.

**Comment by miinki** [comment]
@ptrumpis noob q- would this allow me to create and publish new lenses as well to the project?

**Reply by ptrumpis** [reply]
Every lens you create with Lens Studio 4.36.1 or lower is guranteed to work with this server.
Every lens created with a higher Lens Studio version might need additional [assets ](https://github.com/ptrumpis/snap-camera-server/wiki/Snap-Camera-Assets) or manual fixing with [Lens Tool](https://github.com/ptrumpis/snap-lens-tool).
